mcda/electre/class_assignment.py

```diff
@@ -0,0 +1,203 @@
+from typing import Any, Dict, Union
+
+from mcda.core.aliases import NumericValue
+from mcda.electre.outranking import OutrankingRelation, outranking_relation_marginal
+
+import pandas as pd
+
+
+def assign_tri_rc_class(
+    alternatives: Union[Dict[Any, NumericValue], pd.Series],
+    categories_rank: pd.Series,
+    categories_profiles: pd.Series,
+    outranking_ap: pd.DataFrame,
+    outranking_pa: pd.DataFrame,
+    credibility_ap: pd.DataFrame,
+    credibility_pa: pd.DataFrame,
+) -> pd.Series:
+    """
+    :param alternatives:
+    :param categories_rank:
+    :param categories_profiles:
+    :param outranking_ap:
+    :param outranking_pa:
+    :param credibility_ap:
+    :param credibility_pa:
+    :return:
+    """
+    # sorted categories by ranks - ascending (worst to best)
+    categories = [i for i in categories_rank.sort_values(ascending=False)]
+
+    # list of profiles according to categories
+    profiles = list(sorted(categories_profiles))
+
+    assignments_descending = []
+    assignments_ascending = []
+    for a in alternatives:
```

Comment:
And there's the fun part, cause if the user passes a `pd.Series`, the `a` variable would have the `NumericValue` type, but if it's a `dict`, then `a` would be one of the key's value :))))